File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    if len(prices)<1{
        return 0
    }
    maxprofit := 0
    minPurchase := prices[0]
    for _,price := range(prices){
        if price < minPurchase{
            minPurchase = price
        }else if (price-minPurchase) > maxprofit{
            maxprofit = price - minPurchase
        }
    }
    return maxprofit
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        maxprofit=0
        minprices=prices[0]
        for i in range(1,len(prices)):
            maxprofit=max(maxprofit,prices[i]-minprices)
            minprices=min(minprices,prices[i])
        
        return maxprofit